src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,44 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo | null;
+  setModalOpen: (val: boolean) => void;
+  resetSelectedTodo: (val: Todo | null) => void;
+}
+
+export const TodoModal: React.FC<Props> = ({
+  setModalOpen,
+  todo,
+  resetSelectedTodo,
+}) => {
+  const [fetchedUser, setFetchedUser] = useState<User | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (todo) {
+      getUser(todo.userId)
+        .then(setFetchedUser)
+        .catch(err => {
+          setError(`Error fetching user: ${err.message}`);
+          setFetchedUser(null);
+        });
+    }
+  }, [todo]);
+
+  const handleCloseModal = () => {
+    setModalOpen(false);
```

Comment:
The `setModalOpen` function is expected to take a boolean value, but the current implementation passes `false`. Ensure that the function signature and usage are consistent across the application. If `setModalOpen` is intended to close the modal, consider renaming it to `closeModal` or similar for clarity.

Reply:
> The `setModalOpen` function is expected to take a boolean value, but the current implementation passes `false`. Ensure that the function signature and usage are consistent across the application. If `setModalOpen` is intended to close the modal, consider renaming it to `closeModal` or similar for clarity.

Good point